sources/api/migration/migrations/v1.4.0/registry-mirror-representation/src/main.rs

```diff
@@ -0,0 +1,126 @@
+#![deny(rust_2018_idioms)]
+
+use datastore::{Key, KEY_SEPARATOR};
```

Comment:
We should avoid linking to datastore (or much of anything API-related) in migrations - the code needs to run before and after a version upgrade, during which time the data store code can change, and it makes thinking about the change confusing; what am I linked against?  when am I running?  etc.

I can't think of a better option than copying the valid_character and KEY_SEPARATOR code from datastore into this migration for now.  We do need to know any keys that would be invalid on downgrade so the downgrade doesn't fail; it's better to throw away those settings than keep the user on a new version that's breaking them in other ways.